utils/wasm-gen/src/config/generator.rs

```diff
@@ -0,0 +1,145 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Configs related to instantiation of gear wasm module generators.
+
+use crate::{wasm::PAGE_SIZE as WASM_PAGE_SIZE, SysCallsConfig, SysCallsConfigBuilder};
+
+/// Gear wasm generator config.
+///
+/// This is a carrier for other configs, that can be used separately
+/// in corresponding generators.
+#[derive(Debug, Clone, Default)]
+pub struct GearWasmGeneratorConfig {
+    /// Memory pages config.
+    pub memory_config: MemoryPagesConfig,
+    /// Entry points config.
+    pub entry_points_config: EntryPointsSet,
+    /// Sys-calls generator module config.
+    pub sys_calls_config: SysCallsConfig,
+    /// Flag, signalizing whether recursions
+    /// should be removed from resulting module.
+    pub remove_recursions: bool,
+}
+
+impl GearWasmGeneratorConfig {
+    pub fn default_with_log_info(log: String) -> GearWasmGeneratorConfig {
+        GearWasmGeneratorConfig {
+            sys_calls_config: SysCallsConfigBuilder::new(Default::default(), false)
+                .with_log_info(log)
+                .build(),
+            ..Default::default()
+        }
+    }
+}
+
+/// Memory pages config used by [`crate::MemoryGenerator`].
+#[derive(Debug, Clone, Copy)]
+pub struct MemoryPagesConfig {
+    /// Initial memory size.
+    pub initial_size: u32,
+    /// Optional memory maximum.
+    pub upper_limit: Option<u32>,
+    /// Optional stack end.
+    pub stack_end: Option<u32>,
+}
+
+impl Default for MemoryPagesConfig {
+    fn default() -> Self {
+        Self {
+            initial_size: Self::MAX_VALUE / 2 + 1,
+            upper_limit: Some(Self::MAX_VALUE),
+            // Make value a multiple of WASM_PAGE_SIZE (bytes), but less than min memory.
```

Comment:
This multiplication was a moved from `lib.rs` function which set stach-end. Not sure if it's proper to do that. @grishasobol Can you clarify that?

Reply:
Yes, this was definetely wrong. We had merged invalid code.